### PR TITLE
Changeable role attribute

### LINKED ISSUE #1123: [FEAT REQ] Changeable role attribute (v5)
**Is your feature request related to a problem? Please describe.**
Regarding the tooltip design pattern of the W3C (https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/), a tooltip should "_not receive focus. A hover that contains focusable elements can be made using a non-modal dialog_". 
Currently, I have a use case where my tooltip contains multiple buttons or links. Hence, my tooltip should rather be a non-modal dialog and for dialogs, you usually set `role=dialog` instead of `role=tooltip`.

**Describe the solution you'd like**
Making the role of the tooltip component a changeable property or adding a prop `asDialog` would be very helpful to set the correct role attribute.

### PR DESCRIPTION
**Pull request by mailinhdang**
Closes #1123.

### CODE REVIEW

File: docs/docs/options.mdx [hunk 1]
```diff
@@ -130,3 +130,4 @@ import { Tooltip } from 'react-tooltip';
...
 | `disableStyleInjection` | `boolean` or `'core'`                  | no       | `false`                   | `true` `false` `'core'`                                                                                                           | Whether to disable automatic style injection. Do not set dynamically. Check the [styling page](./examples/styling#disabling-reacttooltip-css) for more details  |
+| `role`                  | `'tooltip'` or `'dialog'`              | no       | `tooltip`                 | `'tooltip'` `'dialog'`                                                                                                            | Set ARIA role, either `tooltip` or `dialog` if the tooltip should contain focusable elements                                                                    |
```
Comment (gabrieljablonski) [inline docs/docs/options.mdx:133]:
```suggestion
| `role`                  | ARIA role              | no       | `tooltip`                 | `'tooltip'` `'dialog'`                                                                                                            | Set ARIA role (`tooltip` by default). Setting to `dialog` is recommended if the tooltip contains focusable elements                                                                    |
```

File: src/components/Tooltip/TooltipTypes.d.ts [hunk 2]
```diff
@@ -153,4 +153,5 @@ export interface ITooltip {
   border?: CSSProperties['border']
   opacity?: CSSProperties['opacity']
   arrowColor?: CSSProperties['backgroundColor']
+  role?: 'tooltip' | 'dialog'
```
Comment (gabrieljablonski) [inline src/components/Tooltip/TooltipTypes.d.ts:156]:
```suggestion
  role?: React.AriaRole
```

File: src/components/TooltipController/TooltipControllerTypes.d.ts [hunk 3]
```diff
@@ -93,6 +93,7 @@ export interface ITooltipController {
   setIsOpen?: (value: boolean) => void
   afterShow?: () => void
   afterHide?: () => void
+  role?: 'tooltip' | 'dialog'
```
Comment (gabrieljablonski) [inline src/components/TooltipController/TooltipControllerTypes.d.ts:96]:
```suggestion
  role?: React.AriaRole
```

**Review by gabrieljablonski — CHANGES_REQUESTED** [review]
Using the generic `AriaRole` type should cover all other possible cases, not only for the `dialog` alternative.

File: src/components/Tooltip/TooltipTypes.d.ts (hunk 2 above)
Comment (mailinhdang) [reply src/components/Tooltip/TooltipTypes.d.ts]: done

File: src/components/TooltipController/TooltipControllerTypes.d.ts (hunk 3 above)
Comment (mailinhdang) [reply src/components/TooltipController/TooltipControllerTypes.d.ts]: done

**Comment by mailinhdang** [comment]
I committed a change with the generic type AriaRole :)
Was thinking about using it before but thought it would be too generic and open for misuse but it's certainly more flexible.

**Review by gabrieljablonski — APPROVED** [review]

**Review by danielbarion — APPROVED** [review]

**Comment by danielbarion** [comment]
Looks great!

Thanks for the contribution :)

**Pull request merged**